joomla/joomla-cms
# Allow to add new options dynamically to select fields

Pull Request for Improvement.
### Summary of Changes

New method `JFormFieldList::addOption()` to allow to add new options to select field dynamicly.
### Testing Instructions
- Code review
- Apply patch on 3.7.x branch
- Add the following code to issues to any backend view with search tools (ex: for articles list, here https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_content/views/articles/view.html.php#L92 )

``` php
$this->filterForm->getField('fullordering', 'list')
    ->addOption('text1', array('value' => 'field ASC'))
    ->addOption('JFEATURED_DESC', array('value' => 'featured DESC'));
```
- Check there are two now options in the searchtools ordering select field
  ![image](https://cloud.githubusercontent.com/assets/9630530/19289201/90386490-9001-11e6-8510-2048ef254f07.png)
### Documentation Changes Required

Update documentation that teaches how to add form fields dynamically.
